tools/crm_resource.c

```diff
@@ -481,12 +481,16 @@ static GOptionEntry location_entries[] = {
       "Lifespan (as ISO 8601 duration) of created constraints (with\n"
       INDENT "-B, -M) see https://en.wikipedia.org/wiki/ISO_8601#Durations)",
       "TIMESPEC" },
-    { "master", 0, G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE, &options.promoted_role_only,
+    { "promoted", 0, G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE, &options.promoted_role_only,
```

Comment:
It might be worth mentioning that --promoted replaces --master here, or in a section in the man page.  I can imagine a user searching for "master" in the help output and not finding anything, and wondering what is going on.

Reply:
I've made --master unhidden and changed its help to "Deprecated: Use --promoted instead".

I've been mulling an idea for a deprecation policy for both API and tool options. For tool options, we could do something like mark an option as deprecated in help for one release series (e.g. 2.1.x), then hide it and print a warning when used in another release series, and finally remove it the series after that.

Reply:
That sounds like a sensible deprecation policy.  It'd be nice to have something in place so we have a plan to get rid of these options.  There's several in crm_mon at least.